addons/mail/static/src/core/web/messaging_menu_patch.js

```diff
@@ -178,4 +186,75 @@ patch(MessagingMenu.prototype, {
         }
         return _t("Failure: %(modelName)s", { modelName: failure.modelName });
     },
+    onClickThread(isMarkAsRead, thread) {
+        if (this.store.discuss.activeTab === "inbox") {
+            if (isMarkAsRead) {
+                thread.needactionMessages[0].setDone();
+                return;
+            } else {
+                thread.highlightMessage = thread.needactionMessages[0];
+            }
+        }
+        super.onClickThread(isMarkAsRead, thread);
+    },
+    get threads() {
+        if (this.store.discuss.activeTab === "inbox") {
+            return this._getInboxTabThreads();
+        }
+        return super.threads;
+    },
+    _getInboxTabThreads() {
+        const searchTerm = cleanTerm(this.store.discuss.searchTerm);
+        let allMessages = [
+            ...this.store.inbox.messages.reverse(),
+            ...this.store.history.messages.reverse(),
+        ];
```

Comment:
As said in previous comment: should not hijack `messages` of many threads and combine that way. This would very likely not set the proper state of message list and compromise loading of messages unless page reloading.

Make sure to reuse message fetch flow of a single thread (including adding a new thread with its specific domain to fetch messages).